mono/metadata/metadata.c

```diff
@@ -2351,49 +2320,58 @@ image_sets_unlock (void)
 static MonoImageSet*
 get_image_set (MonoImage **images, int nimages)
 {
+	/* Cache used only in this function. Protected by image_sets_mutex */
+	static MonoImageSet *mscorlib_image_set;
```

Comment:
Given this function is not part of this change, I think it's best to leave it completely unchanged.